examples/src/picking.ts

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import * as THREE from 'three';
+import CameraControls from 'camera-controls';
+import * as reveal from '@cognite/reveal';
+import * as reveal_threejs from '@cognite/reveal/threejs';
+
+CameraControls.install({ THREE });
+
+async function main() {
+  const modelUrl = new URL(location.href).searchParams.get('model') || '/primitives';
+
+  const pickedNodes: number[] = [];
+  const shading = reveal_threejs.createDefaultShading({
+    color(treeIndex: number) {
+      if (pickedNodes.indexOf(treeIndex) !== -1) {
+        return [255, 255, 0, 255];
+      }
+      return undefined;
+    }
+  });
+
+  const scene = new THREE.Scene();
+  const cadModel = await reveal.createLocalCadModel(modelUrl);
+  const cadNode = new reveal_threejs.CadNode(cadModel, { shading });
+
+  scene.add(cadNode);
+
+  const renderer = new THREE.WebGLRenderer();
+  renderer.setClearColor('#444');
+  renderer.setSize(window.innerWidth, window.innerHeight);
+  document.body.appendChild(renderer.domElement);
+
+  const { position, target, near, far } = cadNode.suggestCameraConfig();
+  const camera = new THREE.PerspectiveCamera(75, window.innerWidth / window.innerHeight, near, far);
+  const controls = new CameraControls(camera, renderer.domElement);
+  controls.setLookAt(position.x, position.y, position.z, target.x, target.y, target.z);
+  controls.update(0.0);
+  camera.updateMatrixWorld();
+  let pickingNeedsUpdate = false;
+  const clock = new THREE.Clock();
+  const render = async () => {
+    const delta = clock.getDelta();
+    const controlsNeedUpdate = controls.update(delta);
+    const sectorsNeedUpdate = await cadNode.update(camera);
+
+    if (controlsNeedUpdate || sectorsNeedUpdate || pickingNeedsUpdate) {
+      renderer.render(scene, camera);
+      pickingNeedsUpdate = false;
+    }
+
+    requestAnimationFrame(render);
+  };
+  render();
+
+  const pickingTarget = new THREE.WebGLRenderTarget(1, 1);
+  const pixelBuffer = new Uint8Array(4);
+
+  const pick = (event: MouseEvent) => {
```

Comment:
Should we have a wrapper for picking for the example (change this to use the wrapper later) or keep the example as is to show a custom way of doing picking?

Reply:
I agree that this should be wrapped in a utility

Reply:
Let us just pretend that I did not forget to push a very relevant commit on Friday and that I just fixed this to adapt to your suggestion :stuck_out_tongue: